incubator/hnc/config/default/kustomization.yaml

```diff
@@ -12,55 +12,59 @@ namePrefix: hnc-
 #commonLabels:
 #  someName: someValue
 
-bases:
-- ../crd
-- ../rbac
-- ../manager
-- ../webhook
-- ../certmanager
 
-patches:
-- manager_image_patch.yaml
   # Protect the /metrics endpoint by putting it behind auth.
   # Only one of manager_auth_proxy_patch.yaml and
   # manager_prometheus_metrics_patch.yaml should be enabled.
-- manager_auth_proxy_patch.yaml
   # If you want your controller-manager to expose the /metrics
   # endpoint w/o any authn/z, uncomment the following line and
   # comment manager_auth_proxy_patch.yaml.
   # Only one of manager_auth_proxy_patch.yaml and
   # manager_prometheus_metrics_patch.yaml should be enabled.
 #- manager_prometheus_metrics_patch.yaml
 
-- manager_webhook_patch.yaml
 
+patchesStrategicMerge:
+- manager_image_patch.yaml
+- manager_auth_proxy_patch.yaml
+- manager_webhook_patch.yaml
 - webhookcainjection_patch.yaml
 
 # the following config is for teaching kustomize how to do var substitution
 vars:
-- name: CERTIFICATE_NAMESPACE # namespace of the certificate CR
+- fieldref:
+    fieldPath: metadata.namespace
+  name: CERTIFICATE_NAMESPACE
   objref:
-    kind: Certificate
     group: certmanager.k8s.io
+    kind: Certificate
+    name: serving-cert
     version: v1alpha1
```

Comment:
Can you reorder these so they match the original order? Note that the first line in each group must have the dash, and maybe try to preserve the comments as well. So the first line should be:

`  - name: CERTIFICATE_NAMESPACE # namespace of the certificate CR`

I'm not sure why `kustomize edit fix` reordered these lines :/ It was probably easier to reorder them than to preserve the existing order. Just make sure that if it _added, changed or removed_ something, you maintain preserve that change.

Reply:
Sure, makes total sense! Not sure why it reordered too.

Reply:
My guess is:

1. It reads the YAML into a Go map
2. It makes the necessary changes
3. It writes the values from the Go maps back as YAML, but with the keys sorted in alphabetical order

This is way easier, but less user-friendly, than trying to preserve the existing line ordering _except_ for the bits that have changed.